Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -493,58 +488,118 @@ WarpX::computeE (amrex::Vector<std::array<std::unique_ptr<amrex::MultiFab>, 3> >
-                    Ex_arr(i,j,k) +=
-                        +(beta_x*beta_x-1)*inv_dx*( phi_arr(i+1,j,k)-phi_arr(i,j,k) )
-                        +beta_x*beta_z*0.25_rt*inv_dz*(phi_arr(i  ,j+1,k)-phi_arr(i  ,j-1,k)
-                                                  + phi_arr(i+1,j+1,k)-phi_arr(i+1,j-1,k));
-                },
-                [=] AMREX_GPU_DEVICE (int i, int j, int k) {
-                    Ez_arr(i,j,k) +=
-                        +beta_z*beta_x*0.25_rt*inv_dx*(phi_arr(i+1,j  ,k)-phi_arr(i-1,j  ,k)
-                                                  + phi_arr(i+1,j+1,k)-phi_arr(i-1,j+1,k))
-                        +(beta_z*beta_z-1)*inv_dz*( phi_arr(i,j+1,k)-phi_arr(i,j,k) );
-                }
-            );
-            ignore_unused(beta_y);
+                amrex::ParallelFor( tbx, tby, tbz,
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ex_arr(i,j,k) +=
+                            +(beta_x*beta_x-1)*0.5_rt*inv_dx*( phi_arr(i+1,j,k)-phi_arr(i-1,j,k) )
+                            +beta_x*beta_z*0.5_rt*inv_dz*(phi_arr(i  ,j+1,k)-phi_arr(i  ,j-1,k));
+                    },
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ey_arr(i,j,k) +=
+                            +beta_x*beta_y*0.5_rt*inv_dx*( phi_arr(i+1,j,k)-phi_arr(i-1,j,k) )
+                            +beta_y*beta_z*0.5_rt*inv_dz*(phi_arr(i  ,j+1,k)-phi_arr(i  ,j-1,k));
+                    },
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ez_arr(i,j,k) +=
+                            +beta_z*beta_x*0.5_rt*inv_dx*(phi_arr(i+1,j  ,k)-phi_arr(i-1,j  ,k))
+                            +(beta_z*beta_z-1)*0.5_rt*inv_dz*( phi_arr(i,j+1,k)-phi_arr(i,j-1,k) );
+                    }
+                );
 #else
-            amrex::ParallelFor( tbz,
-                [=] AMREX_GPU_DEVICE (int i, int j, int k) {
-                    Ez_arr(i,j,k) +=
-                        +(beta_z*beta_z-1)*inv_dz*( phi_arr(i+1,j,k)-phi_arr(i,j,k) );
-                }
-            );
-            ignore_unused(beta_x,beta_y);
+                amrex::ParallelFor( tbx, tby, tbz,
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ex_arr(i,j,k) +=
+                            +(beta_x*beta_z-1)*0.5_rt*inv_dz*( phi_arr(i+1,j,k)-phi_arr(i-1,j,k) );
+                    },
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ey_arr(i,j,k) +=
+                            +beta_y*beta_z*0.5_rt*inv_dz*( phi_arr(i+1,j,k)-phi_arr(i-1,j,k));
+                    },
+                    [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                        Ez_arr(i,j,k) +=
+                            +(beta_z*beta_z-1)*0.5_rt*inv_dz*( phi_arr(i+1,j,k)-phi_arr(i-1,j,k) );
+                    }
+                );
+#endif
+            }
+            else{
```

Comment:
Could you actually do an `else if` that checks that the staggering of the `E` field is the one we expect for a Yee grid (and executes the code below if that is the case) and then an `else` clause that abort by saying that the staggering of `E` is unexpected.